lib/src/showcase.dart

```diff
@@ -35,49 +35,162 @@ import 'tooltip_widget.dart';
 
 class Showcase extends StatefulWidget {
   @override
+
+  /// A key that is unique across the entire app.
   final GlobalKey key;
 
+  /// Target widget that will be showcased or highlighted
   final Widget child;
+
+  /// Represents subject line of target widget
   final String? title;
-  final TextAlign? titleAlignment;
+
+  /// Title alignment with in tooltip widget
+  ///
+  /// Defaults to [TextAlign.start]
+  final TextAlign titleAlignment;
+
+  /// Represents summary description of target widget
   final String? description;
-  final ShapeBorder? shapeBorder;
-  final BorderRadius? radius;
+
+  /// ShapeBorder of the highlighted box when target widget will be showcased.
+  ///
+  /// Note: If [targetBorderRadius] is specified, this parameter is ignored.
+  ///
+  /// Default value is:
+  /// ```dart
+  /// RoundedRectangleBorder(
+  ///   borderRadius: BorderRadius.all(Radius.circular(8)),
+  /// ),
+  /// ```
+  final ShapeBorder targetShapeBorder;
+
+  /// Radius of rectangle box when target widget will be showcased.
+  final BorderRadius? targetBorderRadius;
+
+  /// TextStyle for default tooltip title
   final TextStyle? titleTextStyle;
+
+  /// TextStyle for default tooltip description
   final TextStyle? descTextStyle;
-  final EdgeInsets contentPadding;
+
+  /// Empty space around tooltip content.
+  ///
+  /// Default Value for [Showcase] widget is:
+  /// ```dart
+  /// EdgeInsets.symmetric(vertical: 8, horizontal: 8)
+  /// ```
+  final EdgeInsets tooltipPadding;
+
+  /// Background color of overlay during showcase.
+  ///
+  /// Default value is [Colors.black45]
   final Color overlayColor;
+
+  /// Opacity apply on [overlayColor] (which ranges from 0.0 to 1.0)
+  ///
+  /// Default to 0.75
   final double overlayOpacity;
+
+  /// Custom tooltip when [Showcase.withWidget] is used
+  /// having fixed [height] and [width]
   final Widget? container;
-  final Color showcaseBackgroundColor;
+
+  /// Default tooltip background color
+  ///
+  /// Default to [Colors.white]
+  final Color tooltipBackgroundColor;
+
+  /// Default tooltip text color
+  ///
```

Comment:
Mention, Where this will be applied and and how it differs from titleTextStyle and description text style. As these two also can be used to provide text color.